src/components/guide.tsx/UseIsVisible.tsx

```diff
@@ -0,0 +1,18 @@
+import { useEffect, useState } from "react";
+
+export default function useIsVisible(ref: any) {
```

Comment:
```suggestion
export default function useIsVisible<T extends Element>(
  ref: React.MutableRefObject<T>,
) {
```

Use this as it's better for TS, any can mean that `ref.current` does not exist and can crash the webpage